cider/function/datetime/ExtractFromTime.cpp

```diff
@@ -77,7 +77,11 @@ extract_nanosecond(const int64_t lcltime) {
 }
 
 // First day of epoch is Thursday, so + 4 to have Sunday=0.
-extern "C" RUNTIME_EXPORT ALWAYS_INLINE int64_t extract_dow(const int64_t lcltime) {
+extern "C" RUNTIME_EXPORT ALWAYS_INLINE int64_t extract_dow(const int64_t lcltime,
+                                                            const bool is_null = false) {
+  if (is_null == true) {
```

Comment:
`if (is_null)`

Reply:
same to others.

Reply:
in IR this is ```@extract_month(i64 %10, i1 %8)```, so the argument passed to ```is_null``` is in type ```i1``` 1 bit , instead of ```bool```1 byte. So ```if (is_null)``` is not working and using ==true will convert to compare 0 and 1 which will work. Let me put a comment on this.

Reply:
` extract_dow(const int64_t lcltime, const bool is_null = false)`. Seems the arguments here is `bool` type. Can't it be converted implicitly after 1 bit passed in the method?

Reply:
seems it can't. I tested ```if(is_null)``` and the result is random. Didn't find out the reason why...